audio/engine.asm

```diff
@@ -803,17 +760,14 @@ HandleTrackVibrato:
 	and a
 	jr nz, .subexit
 	; is the extent nonzero?
-	ld hl, wChannel1VibratoExtent - wChannel1
-	add hl, bc
-	ld a, [hl]
+	inc hl
```

Comment:
How about an `assert VIBRATO_DELAY_COUNT + 1 == VIBRATO_EXTENT`? Likewise for other implicit "adjust hl to point to this field" optimizations. (In case you do more and they shift.)

Reply:
I'll merge for now as-is, but more asserts here (and maybe in move_mon.asm, IIRC it also does hli/hld assumptions) would be nice.

Reply:
Left comments on any implicit reference that crosses a field so i can grep for usage

Happens kind of randomly throughout the file though, asserting each instance would add noise from redundant asserts